File: java/src/test/java/com/adaptionsoft/games/trivia/Dice.java
package com.adaptionsoft.games.trivia;

import java.util.Random;

final class Dice {

    private final int diceSize;
    private final Random randomGenerator;

    public Dice(int diceSize, long randomSeed) {
        if (diceSize < 1) throw new RuntimeException("A dice must be larger than one");
        if (randomSeed < 0) throw new RuntimeException("The random seed must be larger than zero");
        this.diceSize = diceSize;
        this.randomGenerator = new Random(randomSeed);

    }

    int roll() {
        return randomGenerator.nextInt(diceSize) + 1;
    }

}

File: kotlin/src/test/kotlin/com/adaptionsoft/games/trivia/Dice.kt
package com.adaptionsoft.games.trivia

import java.lang.RuntimeException
import java.util.*

internal class Dice(diceSize: Int, randomSeed: Long) {
    private val diceSize: Int
    private val randomGenerator: Random

    init {
        if (diceSize < 1) throw RuntimeException("A dice must be larger than one")
        if (randomSeed < 0) throw RuntimeException("The random seed must be larger than zero")
        this.diceSize = diceSize
        randomGenerator = Random(randomSeed)
    }

    fun roll(): Int {
        return randomGenerator.nextInt(diceSize) + 1
    }
}